File: buffer/cpu/src/jvmMain/java/com/wsr/cpu/elementwise/compare/JCompare.java
package com.wsr.cpu.elementwise.compare;

import java.nio.ByteBuffer;

public class JCompare {
    public native void greaterThanD1ToD0(ByteBuffer x, float y, ByteBuffer result);
    public native void greaterThanD1ToD1(ByteBuffer x, ByteBuffer y, ByteBuffer result);

    public native void lessThanD1ToD0(ByteBuffer x, float y, ByteBuffer result);
    public native void lessThanD1ToD1(ByteBuffer x, ByteBuffer y, ByteBuffer result);

    public native void equalsD1ToD0(ByteBuffer x, float y, float absoluteTolerance, float relativeTolerance, ByteBuffer result);
    public native void equalsD1ToD1(ByteBuffer x, ByteBuffer y, float absoluteTolerance, float relativeTolerance, ByteBuffer result);
}

File: buffer/cpu/src/jvmAndroidMain/kotlin/com/wsr/knist/cpu/elementwise/compare/JCompare.kt
package com.wsr.knist.cpu.elementwise.compare

object JCompare {
    external fun greaterThanD1ToD0(x: Long, y: Float, result: Long)
    external fun greaterThanD1ToD1(x: Long, y: Long, result: Long)

    external fun lessThanD1ToD0(x: Long, y: Float, result: Long)
    external fun lessThanD1ToD1(x: Long, y: Long, result: Long)

    external fun equalsD1ToD0(x: Long, y: Float, absoluteTolerance: Float, relativeTolerance: Float, result: Long)
    external fun equalsD1ToD1(x: Long, y: Long, absoluteTolerance: Float, relativeTolerance: Float, result: Long)
}
